pandas/core/frame.py

```diff
@@ -835,15 +835,151 @@ def __len__(self):
 
     def dot(self, other):
         """
-        Matrix multiplication with DataFrame or Series objects
+        Find the dot product of two matrices.
+
+        Using two DataFrame, matrix, or Series objects -
```

Comment:
don't say 'matrix' that is not valid (it works but is not recomneded in any way and is deprecated on the numpy side)., stick to DataFrame/Series.